server/app/lib/actions/fusor/deployment/rhev/ose_launch.rb

```diff
@@ -75,6 +96,74 @@ def ose_launch_completed
           def ose_launch_failed(hostname)
             fail _('OSE Launch failed #{hostname}')
           end
+
+          def generate_ssh_keys (key_type='rsa')
+            deployment = ::Fusor::Deployment.find(input[:deployment_id])
+
+            if File.directory?(get_ssh_key_path)
+              ::Fusor.log.info "====== ssh keys folder '#{get_ssh_key_path}' already exists! skipping key generation"
+            else
+              Utils::Fusor::CommandUtils.run_command("mkdir -p #{get_ssh_key_path}")
+              Utils::Fusor::CommandUtils.run_command("ssh-keygen -f #{get_ssh_key_path}/id_#{key_type} -N '' -t #{key_type}")
+              ::Fusor.log.debug "====== ssh keys have been generated"
+            end
+          end
+
+          def host_exists?(hostname)
+            system("ping -c1 -w1 #{hostname}")
+          end
```

Comment:
I think this is generally useful. Might be put in a utils package/class. Also there exists a ruby gem ```net-ping``` https://rubygems.org/gems/net-ping/versions/1.7.8 though that might cause rpm building which is annoying :)